components/AppBar/src/MDCAppBar.m

```diff
@@ -308,24 +308,36 @@ - (void)viewDidLoad {
                             views:@{kBarStackKey : self.headerStackView}];
   [self.view addConstraints:horizontalConstraints];
 
-  CGFloat topMargin = kPreIOS11StatusBarHeight;
 #if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
   if (@available(iOS 11.0, *)) {
-    // Starting from iOS 11, the top margin should be the actual status bar height.
-    // This is because the flexible header could be smaller in height when in landscape mode
-    // due to the status bar being hidden by default on some devices (like the iPhone X).
-    topMargin = [UIApplication mdc_safeSharedApplication].statusBarFrame.size.height;
+    NSLayoutYAxisAnchor *topAnchor = self.view.safeAreaLayoutGuide.topAnchor;
+    NSLayoutConstraint *topConstraint =
+        [self.headerStackView.topAnchor constraintEqualToAnchor:topAnchor constant:0];
+    topConstraint.active = YES;
+  } else {
+#endif
+  NSLayoutConstraint *topConstraint =
+      [NSLayoutConstraint constraintWithItem:self.headerStackView
+                                   attribute:NSLayoutAttributeTop
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self.view
+                                   attribute:NSLayoutAttributeTop
+                                  multiplier:1
+                                    constant:kPreIOS11StatusBarHeight];
+    topConstraint.active = YES;
```

Comment:
Consider using `topLayoutGuide.bottomAnchor` for iOS 9/10. I know it creates a third condition (iOS 8), but I believe it should adjust for the in-call status bar size.

https://developer.apple.com/documentation/uikit/uiviewcontroller/1621367-toplayoutguide

Reply:
Done. It seems to be working fine with the non-iOS 9 as well though.